platform/linuxbsd/detect.py

```diff
@@ -88,20 +88,24 @@ def configure(env: "SConsEnvironment"):
         # gdb works fine without it though, so maybe our crash handler could too.
         env.Append(LINKFLAGS=["-rdynamic"])
 
-    # Cross-compilation
-    # TODO: Support cross-compilation on architectures other than x86.
-    host_is_64_bit = sys.maxsize > 2**32
-    if host_is_64_bit and env["arch"] == "x86_32":
-        env.Append(CCFLAGS=["-m32"])
-        env.Append(LINKFLAGS=["-m32"])
-    elif not host_is_64_bit and env["arch"] == "x86_64":
-        env.Append(CCFLAGS=["-m64"])
-        env.Append(LINKFLAGS=["-m64"])
-
     # CPU architecture flags.
-    if env["arch"] == "rv64":
+    # TODO: Support cross-compilation on architectures other than x86.
+    if env["arch"] == "x86_64":
+        env.Append(CCFLAGS=["-m64", "-march=x86-64"])
+        env.Append(LINKFLAGS=["-m64", "-march=x86-64"])
+    elif env["arch"] == "x86_32":
+        env.Append(CCFLAGS=["-m32", "-march=i686"])
+        env.Append(LINKFLAGS=["-m32", "-march=i686"])
+    elif env["arch"] == "arm64":
+        env.Append(CCFLAGS=["-march=armv8-a"])
+        env.Append(LINKFLAGS=["-march=armv8-a"])
+    elif env["arch"] == "arm32":
+        env.Append(CCFLAGS=["-march=armv7-a"])
+        env.Append(LINKFLAGS=["-march=armv7-a"])
+    elif env["arch"] == "rv64":
         # G = General-purpose extensions, C = Compression extension (very common).
         env.Append(CCFLAGS=["-march=rv64gc"])
+        env.Append(LINKFLAGS=["-march=rv64gc"])
```

Comment:
This makes it impossible for the user to use a custom `-march` value. For example, compiling with `scons optimize=custom ccflags='-O3 -march=x86-64-v3'` leads to compiler invocations such as `g++ -o core/core_bind.linuxbsd.editor.x86_64.o -c -std=gnu++17 -fno-exceptions -O1 -march=x86-64-v3 -m64 -march=x86-64 [...]`, thus overriding the user-specified flag.

I'd suggest something like this instead:

```suggestion
    if env["arch"] == "x86_64":
        env.Prepend(CFLAGS=["-m64", "-march=x86-64"])
        env.Prepend(CXXFLAGS=["-m64", "-march=x86-64"])
        env.Prepend(LINKFLAGS=["-m64", "-march=x86-64"])
    elif env["arch"] == "x86_32":
        env.Prepend(CFLAGS=["-m32", "-march=i686"])
        env.Prepend(CXXFLAGS=["-m32", "-march=i686"])
        env.Prepend(LINKFLAGS=["-m32", "-march=i686"])
    elif env["arch"] == "arm64":
        env.Prepend(CFLAGS=["-march=armv8-a"])
        env.Prepend(CXXFLAGS=["-march=armv8-a"])
        env.Prepend(LINKFLAGS=["-march=armv8-a"])
    elif env["arch"] == "arm32":
        env.Prepend(CFLAGS=["-march=armv7-a"])
        env.Prepend(CXXFLAGS=["-march=armv7-a"])
        env.Prepend(LINKFLAGS=["-march=armv7-a"])
    elif env["arch"] == "rv64":
        # G = General-purpose extensions, C = Compression extension (very common).
        env.Prepend(CFLAGS=["-march=rv64gc"])
        env.Prepend(CXXFLAGS=["-march=rv64gc"])
        env.Prepend(LINKFLAGS=["-march=rv64gc"])
```

We need to set `CFLAGS` and `CXXFLAGS` individually, because `CCFLAGS` appear after `CFLAGS` and `CXXFLAGS` in the compiler invocation. Just replacing `env.Append` with `env.Prepend` would fix the example SCons command I gave earlier, but it wouldn't fix `scons optimize=custom cflags='-O3 -march=x86-64-v3' cxxflags='-O3 -march=x86-64-v3'`.

Reply:
Good idea, using prepend and ensuring that user flags come after seems like a good idea.

Though, this does make me wonder, should we provide a SCons option to override this? We would have the default be "auto" or similar, then have these defaults if set to auto.